## Metrics - Overview

Useful to evaluate how precise and accurate a ml model is.<br> 
Two types of evaluation metrics are there:<br>

i. Regression metrics<br>
ii. Classification metrics

# Regression metrics

a. Mean squared error(mse) : Average squared difference between predicted and actual values.

MSE = (1/n) * Σ (predicted value - actual value)² <br>

n is the number of data points.<br>
Σ represents the summation of all squared differences.<br>

In [1]:
# importing necessary libraries 
from sklearn.linear_model import LinearRegression 
import seaborn as sb 
from sklearn.model_selection import train_test_split 
from sklearn.metrics import mean_squared_error
# loading the dataset
df = sb.load_dataset('tips') 
# selecting features and target variable 
X = df[['total_bill', 'size']]
y = df['tip'] 

In [3]:
# splitting the dataset into training and testing sets 
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42) 
# creating a linear regression model
model = LinearRegression() 
# fitting the model to the training data
model.fit(X_train, y_train) 
# making predictions on the test data 
y_pred = model.predict(X_test) 
# displaying the made predictions 
y_pred[:5]

array([2.89943693, 1.89520002, 3.85933994, 3.9811207 , 2.27962736])

In [6]:
# calculating the mean squared error 
mse = mean_squared_error(y_test, y_pred) 
mse

0.6485996190543517

<b>Interpretation:</b> The average error or deviation of predicted value from the original values is about 0.65 according to mse(mean squared error) indicating good model performance but still room for improvement.